prompt/prompt.go

```diff
@@ -0,0 +1,260 @@
+package prompt
+
+import (
+	"time"
+
+	v2 "github.com/CyCoreSystems/ari/v2"
+
+	"github.com/CyCoreSystems/ari"
+	"github.com/CyCoreSystems/ari/audio"
+	"github.com/CyCoreSystems/ari/audio/audiouri"
+
+	"golang.org/x/net/context"
+)
+
+// Options describes options for the prompt
+type Options struct {
+	// FirstDigitTimeout is the maximum length of time to wait
+	// after the prompt sequence ends for the user to enter
+	// a response.
+	// If not specified, the default is DefaultFirstDigitTimeout.
+	FirstDigitTimeout time.Duration
+
+	// InterDigitTimeout is the maximum length of time to wait
+	// for an additional digit after a digit is received.
+	// If not specified, the default is DefaultInterDigitTimeout.
+	InterDigitTimeout time.Duration
+
+	// OverallTimeout is the maximum length of time to wait
+	// for a response regardless of digits received.
+	// If not specified, the default is DefaultOverallTimeout.
+	OverallTimeout time.Duration
+
+	// EchoData is the flag for saying each digit as it is input
+	EchoData bool
+
+	// MatchFunc is an optional function which, if supplied, returns
+	// a string and an int.
+	//
+	// The string is allows the MatchFunc to return a different number
+	// to be used as `result.Data`.  This is commonly used for prompts
+	// which look for a terminator.  In such a practice, the terminator
+	// would be stripped from the match and this argument would be populated
+	// with the result.  Otherwise, the original string should be returned.
+	// NOTE: Whatever is returned here will become `result.Data`.
+	//
+	// The int parameter indicates the result of the match, and it should
+	// be one of:
+	//  Incomplete (0) : insufficient digits to determine match.
+	//  Complete (1) : A match was found.
+	//  Invalid (2) : A match could not be found, given the digits received.
+	// If this function returns a non-zero int, then the prompt will be stopped.
+	// If not specified MatchAny will be used.
+	MatchFunc func(string) (string, Status)
+
+	// Which type of word to use when playing '#'
+	SoundHash string // pound or hash
```

Comment:
In the Asterisk base sound library, this is handled by the localization, rather than manually.  It doesn't hurt to have this, but it really should not be needed, when properly used.